core/src/main/java/org/apache/struts/util/ResponseUtils.java

```diff
@@ -35,6 +37,9 @@
  *          $
  */
 public class ResponseUtils {
+	
+	protected static Pattern XML_ENTITY_PATTERN = Pattern.compile("&(?:[a-z\\d]+|#\\d+|#x[a-f\\d]+);");
```

Comment:
See also the suggested test-cases.
```suggestion
	protected static Pattern XML_ENTITY_PATTERN = Pattern.compile("&(?:[a-zA-Z][a-zA-Z\\d]*|#\\d+|#x[a-fA-F\\d]+);");
```